### PR TITLE
Add set_add hint

### LINKED ISSUE #190: Add `set` and `find_element` module hints
From `set`
 - `SET_ADD`

From `find_element`
- `FIND_ELEMENT`
- `SEARCH_SORTED_LOWER`

### PR DESCRIPTION
**Pull request by Juan-M-V**
Resolves: #190

### CODE REVIEW

File: pkg/lambdaworks/lambdaworks.go [hunk 1]
```diff
@@ -86,6 +92,15 @@ func (felt Felt) ToU64() (uint64, error) {
...
 }
 
+// turns a felt to usize
+func (felt Felt) ToUint() (uint, error) {
+	if felt.limbs[0] == 0 && felt.limbs[1] == 0 && felt.limbs[2] == 0 {
+		return uint(felt.limbs[3]), nil
+	} else {
+		return 0, ConversionError(felt, "uint")
+	}
+}
```
Comment (fmoletta) [inline pkg/lambdaworks/lambdaworks.go]: This should call Felt.ToU64() instead of copying its logic

File: pkg/vm/memory/memory.go [hunk 2]
```diff
@@ -150,6 +150,15 @@ func (m *Memory) GetFelt(addr Relocatable) (lambdaworks.Felt, error) {
 	return lambdaworks.FeltZero(), err
 }
 
+// Get a range of memory from the starting relocatable to the starting relocatable + size
+func (m *Memory) GetRange(start Relocatable, size uint) []MaybeRelocatable {
+	var res []MaybeRelocatable
+	for i := uint(0); i < size; i++ {
+		res = append(res, m.Data[start.AddUint(i)])
```
Comment (fmoletta) [inline pkg/vm/memory/memory.go]: Use `Memory.Get` here, fetching from the map directly will yield the zero value if the key is not present, creating a nil `MaybeRelocatable` in the process

**Review by fmoletta — COMMENTED** [review]
Please move find_element_hint_codes.go & set_hint_codes.go to the hint_codes folder
Also, many functions and variables are using snake_case instead of camelCase

File: pkg/vm/memory/memory.go [hunk 3]
```diff
@@ -151,12 +151,16 @@ func (m *Memory) GetFelt(addr Relocatable) (lambdaworks.Felt, error) {
 }
 
 // Get a range of memory from the starting relocatable to the starting relocatable + size
-func (m *Memory) GetRange(start Relocatable, size uint) []MaybeRelocatable {
+func (m *Memory) GetRange(start Relocatable, size uint) ([]MaybeRelocatable, error) {
+	_, ok := m.Data[start.AddUint(size-1)]
+	if !ok {
+		return nil, errors.Errorf("Range out of bounds, cell %v is not present in Memory", start.AddUint(size-1))
+	}
```
Comment (fmoletta) [inline pkg/vm/memory/memory.go]: This check is not robust enough, there can be gaps in the middle of the range

Comment (Juan-M-V) [reply pkg/vm/memory/memory.go]: Resolved: https://github.com/lambdaclass/cairo-vm_in_go/pull/241/commits/67a34851ac08e02fa7c8583fd09fa32eefe50c64

File: pkg/vm/memory/memory.go (hunk 2 above)
Comment (Juan-M-V) [reply pkg/vm/memory/memory.go]: Resolved: https://github.com/lambdaclass/cairo-vm_in_go/pull/241/commits/67a34851ac08e02fa7c8583fd09fa32eefe50c64

**Comment by Juan-M-V** [comment]
> Please move find_element_hint_codes.go & set_hint_codes.go to the hint_codes folder Also, many functions and variables are using snake_case instead of camelCase

Moved hint codes to the hint_codes directory: https://github.com/lambdaclass/cairo-vm_in_go/pull/241/commits/f2ed685608922bf6434fcfc2d0cc1de4ce717009
Changed sake case to camel case: https://github.com/lambdaclass/cairo-vm_in_go/pull/241/commits/1475d36a933098f17c63947c8e3ba48e1de41b25

File: pkg/lambdaworks/lambdaworks.go (hunk 1 above)
Comment (Juan-M-V) [reply pkg/lambdaworks/lambdaworks.go]: Resolved: https://github.com/lambdaclass/cairo-vm_in_go/pull/241/commits/22c3df1ba52ba746f9f35199e4df688826b3fefe

File: pkg/vm/vm_core.go [hunk 4]
```diff
@@ -324,11 +324,13 @@ func (vm *VirtualMachine) DeduceOp1(instruction *Instruction, dst *memory.MaybeR
 				return &dst_rel, dst, nil
 			}
 		case ResMul:
-			dst_felt, dst_is_felt := dst.GetFelt()
-			op0_felt, op0_is_felt := op0.GetFelt()
-			if dst_is_felt && op0_is_felt && !op0_felt.IsZero() {
-				res := memory.NewMaybeRelocatableFelt(dst_felt.Div(op0_felt))
-				return res, dst, nil
+			if op0 != nil && dst != nil {
```
Comment (jrchatruc) [inline pkg/vm/vm_core.go:327]: Is this added check needed? All tests seem to pass when commented out

File: pkg/vm/memory/memory.go (hunk 3 above)
Comment (fmoletta) [reply pkg/vm/memory/memory.go]: I would insist on using Memory.Get, we don't handle temporary segments yet, but once we do Get will also be in chargo of relocating temporary addresses when fetching them from memory and this behaviour will no longer be valid

File: pkg/vm/vm_core.go (hunk 4 above)
Comment (fmoletta) [reply pkg/vm/vm_core.go:327]: It is needed, we should probably add a test for this specific case, but it causes the vm to panick

Comment (jrchatruc) [reply pkg/vm/vm_core.go:327]: Cool, we don't need to do it on this PR but let's add it at some point

File: pkg/vm/memory/memory.go (hunk 3 above)
Comment (Juan-M-V) [reply pkg/vm/memory/memory.go]: My bad, I didn't realize to use memory.Get. Here is the change: https://github.com/lambdaclass/cairo-vm_in_go/pull/241/commits/65d35d124238697d5d83f8043628a149413b7edf

**Review by jrchatruc — APPROVED** [review]

**Review by fmoletta — APPROVED** [review]

**Pull request merged**